packages/map-template/src/components/RouteInstructions/RouteInstructions.jsx

```diff
@@ -18,22 +24,65 @@ function RouteInstructions({ steps, onNextStep, onPreviousStep, originLocation }
     /** Referencing the previous step of each active step */
     const [previous, setPrevious] = useState();
 
-    const [activeStep, setActiveStep] = useState(0);
+    const [activeStep, setActiveStep] = useRecoilState(activeStepState);
+    const [totalSteps, setTotalSteps] = useState();
+
+    const [lastStepZoom, setLastStepZoom] = useState();
+    const [lastStepCenter, setLastStepCenter] = useState();
+
+    const directions = useRecoilValue(directionsResponseState);
+
+    const mapsIndoorsInstance = useRecoilValue(mapsIndoorsInstanceState);
 
-    /*
-     * Make sure to reset active step whenever the steps change.
+    /**
+     * Clone the last step in the directions in order to create a destination step.
+     * Assign the specific travel mode to the destination step and push the destination step at the end of the steps array.
+     *
      */
     useEffect(() => {
-        setActiveStep(0);
+        const lastStep = steps[steps.length - 1];
+        const destinationStep = { ...lastStep }
+        destinationStep.travel_mode = 'DESTINATION'
+        steps.push(destinationStep);
+        setTotalSteps(steps);
     }, [steps]);
 
+    /**
+     * Get the zoom and the center of the last step and the destination step in the directions.
+     *
+     */
+    useEffect(() => {
+        if (activeStep === totalSteps?.length - 1) {
+            const lastStepZoom = mapsIndoorsInstance.getZoom();
+            const lastStepCenter = mapsIndoorsInstance.getMapView().getCenter();
+            setLastStepZoom(lastStepZoom);
+            setLastStepCenter(lastStepCenter);
+
+            if (directions?.destinationLocation) {
+                // Get the destination location
+                const destinationLocation = directions?.destinationLocation;
+
+                // Center the map to the location coordinates.
+                const destinationLocationGeometry = destinationLocation?.geometry.type === 'Point' ? destinationLocation?.geometry.coordinates : destinationLocation?.properties.anchor.coordinates;
+                mapsIndoorsInstance.getMapView().setCenter({ lat: destinationLocationGeometry[1], lng: destinationLocationGeometry[0] });
+
+                // Check if the solution allows the zoom level to be 22.
+                // If yes, set the zoom level to 22, otherwise set it to 21.
+                mapsindoors.services.SolutionsService.getSolution().then(solution => {
```

Comment:
This looks like something I have seen in another part of the code. Can it be extracted into a helper function somewhere?

Reply:
Extracted the code into a helper function ✅